hooks/pre-commit

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+from __future__ import print_function
+from future import standard_library
+standard_library.install_aliases()
+import os
+import urllib.request
+import urllib.parse
+import urllib.error
+from distutils.version import LooseVersion
+import re
+import sys
+import subprocess
+
+pattern = re.compile(r'[."]version["\s":=]+(\d+\.\d+.\d+)')
+version_dict = {}
+version_file = urllib.request.urlopen(
+    "https://raw.githubusercontent.com/weecology/retriever/master/version.txt")
+version_file.readline()
+
+
+def to_string(value):
+    if sys.version_info >= (3, 0, 0):
+        return value.decode("UTF-8")
+    else:
+        return value
+
+
+for lines in version_file:
+    key_values = to_string(lines).split(",")
+    version_dict[key_values[0]] = key_values[1]
+
+
+def get_script_version(file_name):
+    file_name = os.path.split(file_name)[1]
+    return LooseVersion(version_dict[file_name])
+
+
+# check if upstream is added as remote
+remotes = to_string(subprocess.check_output("git remote -v", shell=True))
+
+if u"upstream\thttps://github.com/weecology/retriever.git (fetch)" not in remotes:
+    print("\nPlease first add upstream as a remote."
+          "\ngit remote add upstream https://github.com/weecology/retriever.git")
+    exit(1)
+
+# check if repo is up-to-date with upstream master
+logs_update = to_string(
+    subprocess.check_output("git fetch upstream", shell=True))
+logs_ahead = to_string(
+    subprocess.check_output(
+        "git log --oneline upstream/master..HEAD | wc -l", shell=True))
+logs_behind = to_string(
+    subprocess.check_output(
+        "git log --oneline HEAD..upstream/master | wc -l", shell=True))
+if int(logs_ahead) > 0 and int(logs_behind) > 0:
+    print(
+        "\nCurrent branch is behind upstream/master."
+        "\nPlease update your branch with upstream master before making commits")
+    exit(1)
```

Comment:
I think this line was here mainly to circumvent the problem of having version differences in the same script between our upstream/master and a local repo. 

If we do take this line out, how do you think we should look at the problem where a script is changed in one direction on upstream/master and another locally? Should we put upstream's changes on top of the script the rebase the local changes over it, increasing the script version by the total amount of changes made?